Cargo.toml

```diff
@@ -11,27 +11,22 @@ repository = "https://github.com/matsadler/magnus"
 homepage = "https://github.com/matsadler/magnus"
 documentation = "https://docs.rs/magnus/"
 license = "MIT"
-exclude = [".github", ".gitignore", "build-sys/"]
+exclude = [".github", ".gitignore"]
 
 [workspace]
 members = ["magnus-macros"]
 exclude = ["examples/rust_blank/ext/rust_blank", "examples/custom_exception_ruby/ext/ahriman", "examples/custom_exception_rust/ext/ahriman"]
 
 [features]
-embed = []
-rb-sys-interop = ["rb-sys"]
+embed = ["rb-sys/link-ruby"]
+gem = ["rb-sys/gem"]
+ruby-static = ["rb-sys/ruby-static"]
+ruby-abi-version = ["rb-sys/ruby-abi-version"]
+global-allocator = ["rb-sys/global-allocator"]
```

Comment:
I don't think I want to make `global-allocator` part of Magnus's feature set, there's a bunch of caveats about using it that I don't want to have to support. Anyone who wants to use it should be able to add a dependency on `rb-sys` and enable the feature that way.

Reply:
Would you be ok with renaming it to `experimental-global-allocator`, then a doc outlining the caveats?

Reply:
Reason being: it would be nice to activate this feature without having to duplicate the rb-sys dep, and thus run the build pipeline again.

Reply:
Cargo will only build `rb-sys` once, all the features enabled are added together before building it.

Cargo will usually try and find a version of a dependency that will fulfil all requirements, and only build the dependency multiple times if that's not possible and multiple versions are required.
However because `rb-sys` specifies the `links` field in `Cargo.toml` there can only ever be one crate with the same value for `links` (and that doesn't just prevent other crates with the same links, but other versions of the same crate), so that won't ever happen here.

Reply:
How would version conflicts be settled, i.e. user specifies different version of rb-sys?

Reply:
I added
```toml
rb-sys = { version = "= 0.9.17", default-features = false, features = ["global-allocator"] }
```
to the `rust_blank` example and this is the error you get:

```text
error: failed to select a version for `rb-sys`.
    ... required by package `magnus v0.3.2 (.../magnus)`
    ... which satisfies path dependency `magnus` (locked to 0.3.2) of package `rust_blank v0.2.0 (.../magnus/examples/rust_blank/ext/rust_blank)`
versions that meet the requirements `^0.9.18` are: 0.9.18

the package `rb-sys` links to the native library `rb`, but it conflicts with a previous package which links to `rb` as well:
package `rb-sys v0.9.17`
    ... which satisfies dependency `rb-sys = "=0.9.17"` of package `rust_blank v0.2.0 (.../magnus/examples/rust_blank/ext/rust_blank)`
Only one package in the dependency graph may specify the same links value. This helps ensure that only one copy of a native library is linked in the final binary. Try to adjust your dependencies so that only one package uses the links ='rb-sys' value. For more information, see https://doc.rust-lang.org/cargo/reference/resolver.html#links.

failed to select a version for `rb-sys` which could resolve this conflict
```

Note I had to set the version to `= 0.9.17` to get the error as just `0.9.17` actually means `>= 0.9.17 && < 0.10`.

Reply:
Ok, I just removed it 👍🏻